src/vmm/src/resources.rs

```diff
@@ -28,48 +29,44 @@ use crate::vstate::vcpu::VcpuConfig;
 type Result<E> = std::result::Result<(), E>;
 
 /// Errors encountered when configuring microVM resources.
-#[derive(Debug, derive_more::From)]
+#[derive(Debug, thiserror::Error, derive_more::From)]
```

Comment:
You can use `#[from]` with `thiserror::Error` to cover the functionality of `derive_more::From`, so maybe it makes sense to remove `derive_more::From` here?

Reply:
I tried to. I need to go derive `thiserror::Error` in all variants. I don't think this is part of this PR.

Reply:
I think that this https://github.com/firecracker-microvm/firecracker/issues/3278 will address it